kn/knative-dockerfile-deploy/README.md

```diff
@@ -0,0 +1,97 @@
+# Git Source with Dockerfile to Knative Service
+
+This documents the example Pipeline to build the source code with a Dockerfile in
+the git repo and deploy it as Knative Service.
+It uses [buildah task](../../buildah/README.md) for building the source code and
+[kn task](../README.md) to create or update a Knative Service.
+
+## Prerequisites:
+
+1. Latest Tekton Pipelines [install](https://github.com/tektoncd/pipeline/blob/master/docs/install.md)ed.
+2. `kubectl` CLI [install](https://kubernetes.io/docs/tasks/tools/install-kubectl/)ed.
+3. `tkn` CLI [install](https://github.com/tektoncd/cli#installing-tkn)ed.
+4. User account exists at a container registry (e.g. [quay.io](https://quay.io))
+5. A ServiceAccount to enable access to perform the required operations in the Pipeline,
+   we'll configure one in following section.
+
+### One time setup:
+
+1 - Create a sample namespace `tkn-kn`, we'll reference this namespace in the subsequent operations.
+```bash
+kubectl create namespace tkn-kn
+```
+
+2 - Create a `docker-registry` type secrets for pushing/pulling the built container images.
+```bash
+kubectl create secret docker-registry container-registry --docker-server=<DOCKER-REGISTRY> --docker-username=<USERNAME> --docker-password=<PASSWORD> --docker-email=<EMAIL>
+```
+#### Note:
+- If you are using `docker.io`: Push to (a new private repo at) `docker.io` via buildah and pull via `imagePullSecrets` does not work!
```

Comment:
I am not sue i understand this part : 

pulll via `imagePullSecrets`

maybe i misread it,

Reply:
If we push to a non-existing repo at docker.io via buildah task, it doesn't populate the repoName at docker hub, then, referencing the same repo from docker.io in kn task can't pull the image.
This works though if I pull via podman CLI locally. 
![image](https://user-images.githubusercontent.com/2510184/69707791-6509fb00-1120-11ea-8773-f57fdeaf5f4d.png)

I had a draft issue against catalog buildah task for this, then I discarded this as this doesn't seem to be the case if pushed in same setup to quay.io. May be something on dockerhub side..

Reply:
I think it just need a bit of rephrasing,

Reply:
Rephrased to

- For using `docker.io`: Please [create a new](https://hub.docker.com/repository/create) empty public repository and refer it in subsequent steps.
- For using `quay.io`: No need to create a repository beforehand.